### PR TITLE
Add person studen and teacher Classes

### PR DESCRIPTION
**Pull request by benja27**
-Implement classes and objects in Ruby.
-Implement encapsulation and inheritance with Ruby.
-Run a program using the command line.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @parent_permission = parent_permission
+  end
+
+  # public methods
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  # private methods
```
Comment (algerina) [inline person.rb]: Beside setting up  person, student and teacher classes most of the project's requirements are missing. kindly read the project requirements carefully the work on them.
You are supposed to :

- [ ] Create a class Nameable.

- [ ] Implement a method called correct_name that will raise a NotImplementedError.

- [ ] Turn your Person class to Nameable

- [ ] Make sure that your Person class inherits from Nameable

- [ ] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- [ ] Prepare base Decorator

- [ ] Make sure that it inherits from Nameable.

- [ ] In the constructor assign a nameable object from params to an instance variable.

- [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable.

- [ ] Prepare CapitalizeDecorator and TrimmerDecorator

For the CapitalizeDecorator:

- [ ] Create a class that inherits from the base Decorator class.

- [ ] Implement a method correct_name that capitalizes the output of @nameable.correct_name.

For the TrimmerDecorator:

- [ ] Create a class that inherits from the base Decorator class.

- [ ] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @benja27  ,

Great job so far!  You still have some changes to implement.

May I highlight the following: 
- [x] Your project is professionally documented👏  ✔️
- [x] Linters are passing👏  ✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# class person
+class Person
+  # we set the getter and setter for these elements - id, name and age
+  attr_accessor :id, :name, :age
```
Comment (Shedrack-Sunday) [inline person.rb]: - [ ] From the project requirements, `id` should only be able to be retrieved so only the Getter is defined
- Kindly make this change.

### Instructions:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @benja27        👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the ruby format✔️ 
- Beautiful use of class inheritance  👏    
 
  
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 2 above)
Comment (benja27) [reply person.rb]: done!

**Comment by Olamarx** [comment]
Hi @benja27,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Changes fixed ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**